replica/compaction_group.hh

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "database_fwd.hh"
+#include "sstables/sstable_set.hh"
+
+#pragma once
+
+namespace replica {
+
+// Compaction group is a set of SSTables which are eligible to be compacted together.
+// By this definition, we can say:
+//      - A group contains SSTables from the same table and shard
+//      - Each group can be thought of an isolated LSM tree.
+//      - Time Window strategy has one group per window
```

Comment:
Doesn't that conflict with major compaction?

Reply:
No, because there's no token overlapping among groups, so we can serialize major across groups

Reply:
> there's no token overlapping among groups

How come? Don't all groups in TWCS cover the whole token range and we just segregate the rows across groups based on the time window?

Reply:
Ah you're talking about TWCS. That was just a conceptual statement. Will remove it to not confuse anybody else.

Reply:
Looks like the confusing comment is still here.
I don't know what you mean by "Time Window strategy" here, other than TWCS.